src/main/cpp/src/utilities.hpp

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cudf/types.hpp>
+#include <cudf/utilities/default_stream.hpp>
+#include <cudf/utilities/span.hpp>
+
+#include <rmm/cuda_stream_view.hpp>
+#include <rmm/device_uvector.hpp>
+
+namespace spark_rapids_jni {
+
+/**
+ * @brief Bitwise-or an array of equally-sized bitmask buffers into a single output buffer
```

Comment:
What is the difference of this vs `cudf::bitmask_or(table_view, ...)`?

Reply:
This works on raw device buffers, not tables/columns.

Reply:
But are they doing the same thing? If so, can we create dummy columns with the given bitmasks then just call `cudf::bitmask_or`?

Reply:
It's close, but the work to get everything wrapped up would end up being pretty much the same as just setting up the array of pointers and calling thrust like we're doing here. It would also involve more memory allocation (in terms of number of allocations).

Reply:
Also I'd say that philosophically we're already down below the abstraction level of columns and tables. We've playing around with buffers.  It's somewhat odd to jump back up to the realm of things like columns and tables when the data we've got really isn't the same (meaning the columns we'd be creating wouldn't even have data in them.  We be just faking it so we can manipulate the bitmasks).